src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -108,37 +111,48 @@ void BATT_SMBUS::RunImpl()
 	// Read current.
 	ret |= _interface->read_word(BATT_SMBUS_CURRENT, result);
 
-	new_report.current_a = (-1.0f * ((float)(*(int16_t *)&result)) / 1000.0f);
+	new_report.current_a = (-1.0f * ((float)(*(int16_t *)&result)) / 1000.0f) * _c_mult;
 	new_report.current_filtered_a = new_report.current_a;
 
 	// Read average current.
 	ret |= _interface->read_word(BATT_SMBUS_AVERAGE_CURRENT, result);
 
-	float average_current = (-1.0f * ((float)(*(int16_t *)&result)) / 1000.0f);
+	float average_current = (-1.0f * ((float)(*(int16_t *)&result)) / 1000.0f) * _c_mult;
 
 	new_report.average_current_a = average_current;
 
 	// If current is high, turn under voltage protection off. This is neccessary to prevent
 	// a battery from cutting off while flying with high current near the end of the packs capacity.
 	set_undervoltage_protection(average_current);
 
-	// Read run time to empty.
+	// Read run time to empty (minutes).
 	ret |= _interface->read_word(BATT_SMBUS_RUN_TIME_TO_EMPTY, result);
 	new_report.run_time_to_empty = result;
 
-	// Read average time to empty.
+	// Read average time to empty (minutes).
 	ret |= _interface->read_word(BATT_SMBUS_AVERAGE_TIME_TO_EMPTY, result);
 	new_report.average_time_to_empty = result;
 
 	// Read remaining capacity.
 	ret |= _interface->read_word(BATT_SMBUS_REMAINING_CAPACITY, result);
 
-	// Calculate remaining capacity percent with complementary filter.
-	new_report.remaining = 0.8f * _last_report.remaining + 0.2f * (1.0f - (float)((float)(_batt_capacity - result) /
-			       (float)_batt_capacity));
+	// Calculate total discharged amount in mah.
+	new_report.discharged_mah = _batt_capacity - (float)result * _c_mult;
+
+	// Read Relative SOC.
+	ret |= _interface->read_word(BATT_SMBUS_RELATIVE_SOC, result);
+
+	// Normalize 0.0 to 1.0
+	new_report.remaining = (float)result / 100.0f;
+
+	// Read SOH
+	// TODO(hyonlim): this value can be used for battery_status.warning mavlink message.
+	ret |= _interface->read_word(BATT_SMBUS_STATE_OF_HEALTH, result);
+	new_report.state_of_health = result;
 
-	// Calculate total discharged amount.
-	new_report.discharged_mah = _batt_startup_capacity - result;
+	// Read Max Error
+	ret |= _interface->read_word(BATT_SMBUS_MAX_ERROR, result);
+	new_report.max_error = result;
```

Comment:
This only needs to be read once on boot. max_error doesn't change through a discharge cycle.